osu.Framework/IO/Network/WebRequest.cs

```diff
@@ -238,7 +238,12 @@ public byte[] GetResponseData()
         public async Task PerformAsync(CancellationToken cancellationToken = default)
         {
             if (Completed)
+            {
+                if (Aborted)
+                    throw new OperationCanceledException($"The {nameof(WebRequest)} has been aborted.");
```

Comment:
This can pass the cancellation token:

```suggestion
                    throw new OperationCanceledException($"The {nameof(WebRequest)} has been aborted.", cancellationToken);
```

Not sure it does much except net style points, though. In theory the runtime is supposed to be checking whether the exception's token is the same as the task's and then decide whether the state of the task should be `Canceled` or `Faulted` (would be `Canceled` with the proposed change), but I can't actually observe this happening.

Reply:
You're probably not observing it happening because you'd want to use the cancellation token of the request itself (i.e. the ones inside `internalPerform()`).

It would be nice to have, but I think it's fine not having it for now, as you'd still get a faulted if it's aborted during the request... Notice how `internalPerform()` calls `Complete()` on abort with an exception, which is then rethrown outwards from `Complete()`.